graphdatascience/aura_sessions.py

```diff
@@ -0,0 +1,129 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import List, Optional, Tuple
+
+from neo4j import GraphDatabase
+
+from graphdatascience import GraphDataScience
+from graphdatascience.aura_api import AuraApi, InstanceDetails
+from graphdatascience.query_runner.aura_db_arrow_query_runner import (
+    AuraDbConnectionInfo,
+)
+
+
+@dataclass
+class SessionInfo:
+    name: str
+
+
+class AuraSessions:
+    GDS_SESSION_NAME_PREFIX = "gds-session-"
+
+    def __init__(
+        self,
+        db_credentials: AuraDbConnectionInfo,
+        aura_api_client_auth: Tuple[str, str],
+        tenant_id: Optional[str] = None,
+    ) -> None:
+        self._db_credentials = db_credentials
+        self._aura_api = AuraApi(
+            tenant_id=tenant_id, client_id=aura_api_client_auth[0], client_secret=aura_api_client_auth[1]
+        )
+
+    # TODO wrapper around it for closable?
+    def create_gds(self, session_name: str, session_password: str) -> GraphDataScience:
+        if len(session_password) < 8:
+            raise ValueError("Password must be at least 8 characters long.")
+
+        if session_name in [session.name for session in self.list_sessions()]:
+            raise RuntimeError(f"Session with name `{session_name}` already exists.")
+
+        create_details = self._aura_api.create_instance(AuraSessions._instance_name(session_name))
+        self._aura_api.wait_for_instance_running(create_details.id)
+
+        gds_user = create_details.username
```

Comment:
Good practice to prefix the usernames with where they belong too. I guess it can get very confusing with the number of credentials and usernames we are dealing with :)